rest/routing.go

```diff
@@ -28,6 +29,56 @@ const docRegex = "[^_/][^/]*"
 //  - A user name with an escaped "/" character
 var docWithSlashPathRegex *regexp.Regexp
 
+// Permission stores the name of a permission along whether it is database scoped. This is used to later obtain a
+// formatted permission string for checking.
+type Permission struct {
+	PermissionName string
+	DatabaseScoped bool
+}
```

Comment:
Maybe move to a new `routing_permissions.go` and move the list of permission variables up to the top so they're easier to find?